snuba/query/validation/functions.py

```diff
@@ -0,0 +1,31 @@
+import logging
+from typing import Sequence
+
+from snuba.query.data_source import DataSource
+from snuba.query.expressions import Expression
+from snuba.query.functions import is_valid_global_function
+from snuba.query.validation import FunctionCallValidator, InvalidFunctionCall
+
+logger = logging.getLogger(__name__)
+
+
+class AllowedFunctionValidator(FunctionCallValidator):
+    """
+    Validates that the function itself is allowed. This does not validate
+    that the function is correctly formed.
+    """
+
+    def __init__(self, enforce: bool = False):
+        self.__enforce = enforce
+
+    def validate(
+        self, func_name: str, parameters: Sequence[Expression], data_source: DataSource
+    ) -> None:
+
+        if is_valid_global_function(func_name):
+            return
+
+        if self.__enforce:
+            raise InvalidFunctionCall(f"Invalid function name: {func_name}")
+        else:
+            logger.warning(f"Invalid function name: {func_name}", exc_info=True)
```

Comment:
@fpacifici @evanh in terms of using this to figure out what functions I might have missed from the list, I don't know if this is going to be too noisy if I miss a really simple one that's in like a very common query

Reply:
This would be ok but it would create a lot of different issues at first. 
I would advise to use a datadog metric and pass the function name as a tag.

Reply:
I second this.